jobclient/src/main/java/com/twosigma/cook/jobclient/Constraint.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) Two Sigma Open Source, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.twosigma.cook.jobclient;
+
+import com.google.common.base.Preconditions;
+import org.json.JSONArray;
+
+/**
+ * A simple class to represent a constraint.
+ *
+ * A constraint has 3 fields: an attribute, an operator, and a value.
+ * An example of constraints could be: "host","EQUALS", "foo.bar.com".
+ *
+ * For now, only "EQUALS" operator is supported by Cook. One could construct
+ * a constraint with an "EQUALS" operator by
+ *
+ * <pre>
+ *  Constraint c = Constraint.EQUALS.apply("host", "foo.bar.com");
+ * </pre>
+ */
+final public class Constraint {
+
+    public enum Operator {
+        EQUALS("EQUALS");
+
+        Operator(String name) {
+        }
+
+        /**
+         * Returns a constraint by "applying" this operator to the given attribute and value.
+         *
+         * @param attribute The constraint attribute
+         * @param value     The constraint value
+         * @return a constraint.
+         */
+        public Constraint apply(String attribute, String value) {
+            return new Constraint(attribute, this, value);
+        }
+
+        /**
+         * Parse an operator from its string representation.
+         *
+         * @param op specifies a string representation of operator.
+         * @return an operator for the specified name.
+         */
+        public static Constraint.Operator fromString(String op) {
+            return Enum.valueOf(Constraint.Operator.class, op.trim().toUpperCase());
+        }
+    }
+
+    private String _attribute;
+    private Operator _operator;
+    private String _value;
+
+    Constraint(String attribute, Operator operator, String value) {
+        Preconditions.checkNotNull(attribute);
+        Preconditions.checkNotNull(operator);
+        Preconditions.checkNotNull(value);
+        _attribute = attribute;
+        _operator = operator;
+        _value = value;
+    }
+
+    Constraint(String attribute, String operator, String value) {
```

Comment:
This looks like a duplicate of the constructor on line 69

Reply:
Yes, all package level constructors. I should call the other one internally